package/MDAnalysis/coordinates/PDB.py

```diff
@@ -391,14 +391,25 @@ def _read_frame(self, frame):
             elif line[:6] == 'CRYST1':
                 # does an implicit str -> float conversion
                 try:
-                    self.ts._unitcell[:] = [line[6:15], line[15:24],
-                                            line[24:33], line[33:40],
-                                            line[40:47], line[47:54]]
+                    cell_dims = np.array([line[6:15], line[15:24],
+                                  line[24:33], line[33:40],
+                                  line[40:47], line[47:54]], 
+                                  dtype=np.float32)  
                 except ValueError:
                     warnings.warn("Failed to read CRYST1 record, "
                                   "possibly invalid PDB file, got:\n{}"
                                   "".format(line))
-
+                else:
+                    if (np.array(cell_dims, dtype=np.float32) == 
```

Comment:
You don't need to make a `np.array` from cell_dims, it already is one (see your recent change above).